### PR TITLE
chore(README.md, package.json): Update npm script documentation

### PR DESCRIPTION
**Pull request by martinsik**
I've noticed there were some leftovers from #2851.

I added more comments to appropriate lines in code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -153,11 +153,10 @@ The build and test structure is fairly primitive at the moment. There are variou
 - build_amd: Transpiles the ES6 files from `dist/es6` to `dist/amd`
 - build_global: Transpiles/Bundles the CommonJS files from `dist/cjs` to `dist/global/Rx.js`
 - build_all: Performs all of the above in the proper order.
-- build_test: builds ES6, then CommonJS, then runs the tests with `jasmine`
 - build_perf: builds ES6, CommonJS, then global, then runs the performance tests with `protractor`
 - build_docs: generates API documentation from `dist/es6` to `dist/docs`
 - build_cover: runs `istanbul` code coverage against test cases
-- test: runs tests with `jasmine`, must have built prior to running.
+- test: runs tests with `mocha`
```
Comment (martinsik) [inline README.md]: It's not using `jasmine` any more and the tests don't need to be build first.

File: README.md [hunk 2]
```diff
@@ -172,7 +171,8 @@ npm run build_all
 ## Performance Tests
 
 Run `npm run build_perf` or `npm run perf` to run the performance tests with `protractor`.
-Run `npm run perf_micro` to run micro performance test benchmarking operator.
+
+Run `npm run perf_micro [operator]` to run micro performance test benchmarking operator.
```
Comment (martinsik) [inline README.md]: I think this was intended to be on a new line (just Markdown requires two new lines). I also added `[operator]` to make it obvious you can run performance tests on a single operator. I'm not sure the description here https://github.com/ReactiveX/rxjs/blob/master/CONTRIBUTING.md#micro is still accurate.

Also this section should probably mention that you need to run `npm run build_all` first.

File: package.json [hunk 3]
```diff
@@ -65,9 +64,8 @@
     "lint": "Run lint against everything",
     "perf": "Run macro performance benchmark",
     "perf_micro": "Run micro performance benchmark",
-    "test_mocha": "Execute mocha test runner against existing test spec build",
     "test_browser": "Execute mocha test runner on browser against existing test spec build",
-    "test": "Clean up existing test spec build, build test spec and execute mocha test runner",
+    "test": "Execute mocha test runner",
```
Comment (martinsik) [inline package.json]: It doesn't cleanup anything and doesn't need to be build prior.

File: package.json [hunk 4]
```diff
@@ -117,7 +115,7 @@
     "test:cover": "cross-env TS_NODE_FAST=true nyc npm test",
     "test:circular": "dependency-cruise --validate .dependency-cruiser.json -x \"^node_modules\" src",
     "tests2png": "tsc && mkdirp tmp/docs/img && mkdirp spec-js/support && shx cp spec/support/*.opts spec-js/support/ && mocha --opts spec/support/tests2png.opts spec-js",
-    "watch": "watch \"echo triggering build && npm run build_test && echo build completed\" src -d -u -w=15"
+    "watch": "watch \"echo triggering build && npm run test && echo build completed\" src -d -u -w=15"
```
Comment (martinsik) [inline package.json]: I'm not sure if `watch` is supposed to always run all tests.

File: README.md [hunk 5]
```diff
@@ -153,11 +153,10 @@ The build and test structure is fairly primitive at the moment. There are variou
 - build_amd: Transpiles the ES6 files from `dist/es6` to `dist/amd`
 - build_global: Transpiles/Bundles the CommonJS files from `dist/cjs` to `dist/global/Rx.js`
 - build_all: Performs all of the above in the proper order.
-- build_test: builds ES6, then CommonJS, then runs the tests with `jasmine`
```
Comment (martinsik) [inline README.md]: `build_test` doesn't exist any more.

**Comment by benlesh** [comment]
Thank you, @martinsik .... sorry for the slow merge

**Pull request merged**